packages/graphback-core/src/runtime/CRUDService.ts

```diff
@@ -72,40 +79,51 @@ export class CRUDService<Type = any> implements GraphbackCRUDService<Type>  {
     return result;
   }
 
-  //tslint:disable-next-line: no-reserved-keywords
-  public async delete(data: Type, context: GraphbackContext): Promise<Type> {
-    if (this.crudOptions.subDelete) {
-      context.graphback.options.selectedFields = [];
+  public async delete(data: Type, context?: GraphbackContext, info?: GraphQLResolveInfo): Promise<Type> {
+    let selectedFields: string[];
+    if (info && !this.crudOptions.subDelete) {
+      selectedFields = getSelectedFieldsFromResolverInfo(info, this.model);
     }
 
-    const result = await this.db.delete(data, context);
+    const result = await this.db.delete(data, selectedFields);
 
     if (this.pubSub && this.crudOptions.subDelete) {
-      const topic = this.subscriptionTopicMapping(GraphbackOperationType.DELETE, this.modelName);
+      const topic = this.subscriptionTopicMapping(GraphbackOperationType.DELETE, this.model.graphqlType.name);
       const payload = this.buildEventPayload('deleted', result);
       await this.pubSub.publish(topic, payload);
     }
 
     return result;
   }
 
-  public findOne(args: Partial<Type>, context: GraphbackContext): Promise<Type> {
-    return this.db.findOne(args, context)
-  }
+  public findOne(args: Partial<Type>, context?: GraphbackContext, info?: GraphQLResolveInfo): Promise<Type> {
+    let selectedFields: string[];
+    if (info) {
+      selectedFields = getSelectedFieldsFromResolverInfo(info, this.model);
+    }
 
-  public async findBy(filter: QueryFilter<Type>, context: GraphbackContext, page?: GraphbackPage, orderBy?: GraphbackOrderBy): Promise<ResultList<Type>> {
-    let count: number;
+    return this.db.findOne(args, selectedFields);
+  }
 
-    if (context.graphback.options?.aggregations?.count) {
-      count = await this.db.count(filter);
+  public async findBy(args?: FindByArgs, context?: GraphbackContext, info?: GraphQLResolveInfo): Promise<ResultList<Type>> {
+    let selectedFields: string[];
+    let requestedCount: boolean = false;
+    if (info) {
+      selectedFields = getSelectedFieldsFromResolverInfo(info, this.model, 'items');
+      requestedCount = getResolverInfoFieldsList(info).some((field: string) => field === "count");
     }
```

Comment:
What  will be the resulting query when a `oneToOne` or `manyToOne` relationship is being resolved? 
- https://github.com/aerogear/graphback/blob/6a6deec2c5189456ab08c2fd10f6315ec2dd26fe/packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts#L718

?

Reply:
Damn, that looks wrong. Need to check it out further.

Reply:
Yeah this is something I overlooked. This assumes the path will always fetch fields at the items node.

Reply:
The only reliable way I can think of doing this is by adding a `path` argument to the findBy method:

```ts
findBy(args?: FindByArgs, context?: GraphbackContext, info?: GraphQLResolveInfo, path?: string): Promise<ResultList<Type>>;
```

I think there is no way to be 100% sure that the path is correct if we try to do this dynamically in the CRUD method. path would be opt in, and in most cases would be undefined, except for the `find<T>` resolver.

Reply:
That's what I was thinking of also.

Reply:
And I think the only place that will need changing (to pass the `path` arg) is for `findXXXs` resolver.

Reply:
Yep you are correct I found this while testing it out.

Reply:
Cool!

Reply:
Fixed in https://github.com/aerogear/graphback/pull/1944/commits/d425e44b73a58ecc768435ddee28e79e364a800b